NYUCodebase/Game/GameState.cpp

```diff
@@ -467,6 +479,19 @@ void GameState::Update(float elapsed) {
         }
     }
     
+    // Check if player has key and is at door
+    int x = map -> worldToTileCoordX(player -> position.x);
+    int y = map -> worldToTileCoordY(player -> position.y);
+    if (keyCollected && x < map -> mapWidth &&
+        y < map -> mapHeight && map -> mapData [y][x] - 1 == 127) {
+        level++;
```

Comment:
This is more of a future thing, but can you also check that the level isn't 3 before incrementing? Once we reach level 3 and win, it should transition to some kind of win screen instead of going on to level 4. You can leave an empty if statement and stub it out for later.